kartothek/io/iter.py

```diff
@@ -196,6 +192,7 @@ def update_dataset_from_dataframes__iter(
     sort_partitions_by=None,
     secondary_indices=None,
     factory=None,
+    metadata_storage_format=DEFAULT_METADATA_STORAGE_FORMAT,
```

Comment:
We've been frequently discussing dropping this kwarg once and for all. I would suggest to not introduce it now for functions where it's been forgotten so far.

Reply:
It's required to not break the `store_` signature